### ISSUE TITLE
Rest API confusion.

### ISSUE THREAD

**Issue by angelfor3v3r** [cpacker/MemGPT#998]
**Describe the bug**
I'm trying to create software that talks to the Rest API endpoints. My goal is to have multiple users, with unique UUIDS or names and each of them can talk to the AI (while each has their own memory).

Most of this is fine, I can easily keep track of everything myself and just pass it to the API endpoints. My assumption is I can pass a userid and an agent id and the backend will automagically look stuff up for the user.

My containers and everything seem to work fine, my container name is `api` so I'm able to reach the API with `http://api:9200/api`.

My initial idea was to call the `/api/agents/config` endpoint like so:
```python
headers = {
    "Accept": "application/json",
    "Content-Type": "application/json"
}

body = {
    "user_id": "foobar",
    "config": {
        "model": "gpt-3.5-turbo",
        "preset": "memgpt_chat",
        "openai_key": "my-api-key-here",
    }
}

my_agent_uuid = None

async with aiohttp.ClientSession().post("http://api:9200/api/agents", headers=headers, json=body) as resp:
    if resp.status == 200:
        data = await resp.json()

        if data and "agent_id" in data:
            my_agent_uuid = data["agent_id"]

print(f"my_agent_uuid = {my_agent_uuid}")
```

This doesn't work and errors with: `{'detail': 'badly formed hexadecimal UUID string'}`.

To make things more confusing, I can't set things like `persona` or `human` in my post request, because that requires a file to be placed in the respective directories (it would be nice to automate this with the API). I could technically hardcode this and mount a directory, and write a persona file... but this seems hacky to me. I also have no idea if my `openai_key` has to go into the `config` json key or the `config -> llm_config` key.

My next idea was to pass along something like so:
```python
my_uuid = str(uuid.uuid4()) # Random UUID.

# Save UUID later, look it up by user myself.

body = {
    "user_id": my_uuid
    "config": {
        "model": "gpt-3.5-turbo",
        "preset": "memgpt_chat",
        "openai_key": "my-api-key-here",
    }
}
```

This errors with: ` {'detail': 'User user_id=09bfab1c-a0d6-45d0-879b-35ad2e839710 does not exist'}` (or whatever UUID was generated).

I'm very new to this type of stuff, and I'm fairly new to python. Perhaps this is the wrong library for my needs? I apologize if so, and help would be appreciated!

**Please describe your setup**
- [ ] How did you install memgpt?
  - I created a docker container myself and run a command like so: `CMD ["memgpt", "server", "--type", "rest", "--host", "0.0.0.0", "--port", "9200"]`
- [ ] Describe your setup
  - My host is Windows but my Docker container is Debian.
  - I'm using version [0.3.2](https://github.com/cpacker/MemGPT/releases/tag/0.3.2)

**Screenshots**
N/A

**Additional context**
N/A

**MemGPT Config**
Please attach your `~/.memgpt/config` file or copy past it below.

```
$ cat config
[defaults]
preset = memgpt_chat
persona = sam_pov
human = basic

[model]
model = gpt-4
model_endpoint = https://api.openai.com/v1
model_endpoint_type = openai
context_window = 8192

[embedding]
embedding_endpoint_type = openai
embedding_endpoint = https://api.openai.com/v1
embedding_model = text-embedding-ada-002
embedding_dim = 1536
embedding_chunk_size = 300

[archival_storage]
type = chroma
path = /home/gpt-bot-api/.memgpt/chroma

[recall_storage]
type = sqlite
path = /home/gpt-bot-api/.memgpt

[metadata_storage]
type = sqlite
path = /home/gpt-bot-api/.memgpt

[version]
memgpt_version = 0.3.2

[client]
anon_clientid = 000000000000000000000242c0a8a002
```

**Comment by sarahwooders** [comment]
@angelfor3v3r thanks for posting this and for trying out MemGPT! We're still in-progress with adding multi-user support to the server, so I think the server is still missing functionality to create a user (which is why the user ID you're providing is missing). @cpacker is about to add a PR that supports user authentication and identify tracking, so once we do that we'll add and document user creation. We should have the functionality you need in 1-2 days.

**Comment by angelfor3v3r** [comment]
> @angelfor3v3r thanks for posting this and for trying out MemGPT! We're still in-progress with adding multi-user support to the server, so I think the server is still missing functionality to create a user (which is why the user ID you're providing is missing). @cpacker is about to add a PR that supports user authentication and identify tracking, so once we do that we'll add and document user creation. We should have the functionality you need in 1-2 days.

Thanks! Do you know if the server API changes will include a way to set persona/human for the AI? The only way I can think of currently is mounting the `.memgpt` folder (for Docker, in my case) and writing a txt file manually.

I'll be sure to test it with the new changes!

**Comment by sarahwooders** [comment]
@angelfor3v3r -- thanks for your patience, we just merged #1014  which adds support for an admin API that can create users that authenticate through the server, and we also have an API call for editing persona/human for the agent https://memgpt.readme.io/reference/update_agent_memory_api_agents_memory_post

You can check out the API docs here https://memgpt.readme.io/reference/api -- it's very new so would love to get your feedback!

**Comment by sarahwooders** [comment]
I'll close this issue for now, but please make a new issue if you run into problems.
